urc_tf/src/world_frame.cpp

```diff
@@ -0,0 +1,96 @@
+#include "world_frame.hpp"
+
+namespace world_frame {
+WorldFrameBroadcaster::WorldFrameBroadcaster(const rclcpp::NodeOptions &options)
+    : Node("world_frame", options) {
+  tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(this);
+
+  gps_subscriber = create_subscription<sensor_msgs::msg::NavSatFix>(
+      "/fix", rclcpp::SystemDefaultsQoS(),
+      [this](const sensor_msgs::msg::NavSatFix msg) { GPSCallback(msg); });
+
+  waypoint_subscriber = create_subscription<urc_msgs::msg::Waypoint>(
+      "/waypoint", rclcpp::SystemDefaultsQoS(),
+      [this](const urc_msgs::msg::Waypoint msg) { WaypointCallback(msg); });
+
+  set_base_subscriber = create_subscription<std_msgs::msg::Bool>(
+      "/set_base", rclcpp::SystemDefaultsQoS(),
+      [this](const std_msgs::msg::Bool msg) { SetBaseCallback(msg); });
+
+  imu_subscriber = create_subscription<sensor_msgs::msg::Imu>(
+      "/imu/data", rclcpp::SystemDefaultsQoS(),
+      [this](const sensor_msgs::msg::Imu msg) { IMUCallback(msg); });
+
+  baseStationLat = -1000;
+  baseStationLng = -1000;
+  baseStationAlt = -1000;
+}
+
+void WorldFrameBroadcaster::SetBaseCallback(const std_msgs::msg::Bool &msg) {
+  this->baseStationLat = currentLat;
+  this->baseStationLng = currentLng;
+  this->baseStationAlt = currentAlt;
+}
+
+void WorldFrameBroadcaster::IMUCallback(const sensor_msgs::msg::Imu &msg) {
+  this->imu_x = msg.orientation.x;
+  this->imu_y = msg.orientation.y;
+  this->imu_z = msg.orientation.z;
+  this->imu_w = msg.orientation.w;
+}
+
+void WorldFrameBroadcaster::GPSCallback(
+    const sensor_msgs::msg::NavSatFix &msg) {
+  this->currentLat = msg.latitude;
+  this->currentLng = msg.longitude;
+  this->currentAlt = msg.altitude;
+
+  if (baseStationLat < -200 || baseStationLng < -200) {
```

Comment:
Also print an error here using `RCLCPP_ERROR(...)`

Reply:
Also, this way of checking if the base station's lat/lng is set is a little awkward. Maybe create a boolean that is set to true when this is set? Then you can just do:
```cpp
if (!isBaseStationCoordinatesSet) {
    return;
}
```